pipelines/shopgate.checkout.processShipping.v1.json

```diff
@@ -0,0 +1,97 @@
+{
+  "version": "1",
+  "pipeline": {
+    "id": "shopgate.checkout.processShipping.v1",
+    "public": false,
+    "input": [
+      {"key": "checkout", "id": "1"}
+    ],
+    "output": [
+      {"key": "checkout", "id": "100"}
+    ],
+    "steps": [
+      {
+        "_comment": "Copy checkout object for modification.",
+
+        "type": "extension",
+        "id": "@shopgate/util",
+        "path": "@shopgate/util/util/merge.js",
+        "input": [
+          {"key": "first", "id": "1"}
+        ],
+        "output": [
+          {"key": "data", "id": "100"}
+        ]
+      },
+
+      {
+        "_comment": "Request the currently selected shipping method from the incoming checkout.",
+
+        "type": "pipeline",
+        "id": "shopgate.checkout.getShippingMethod.v1",
+        "input": [
+          {"key": "checkout", "id": "100"}
+        ],
+        "output": [
+          {"key": "shippingMethod", "id": "121", "optional": true}
```

Comment:
At this point should not be optional anymore. Should be mandatory

Reply:
Not really. that is to be decided later in the (yet missing) validation. you don't necessarily need a shipping method for every order that might be created.